core/src/layers/otelmetrics.rs

```diff
@@ -334,101 +279,207 @@ impl<A: Access> Layer<A> for OtelMetricsLayer {
+            }
+            observe::MetricValue::HttpResponseDurationSeconds(v) => self
+                .http_response_duration_seconds
+                .record(v.as_secs_f64(), &attributes),
+            observe::MetricValue::HttpConnectionErrorsTotal => {
+                self.http_connection_errors_total.add(1, &attributes)
+            }
+            observe::MetricValue::HttpStatusErrorsTotal => {
+                self.http_status_errors_total.add(1, &attributes)
+            }
+        }
     }
 }
 
 impl OtelMetricsInterceptor {
-    fn create_attributes(
-        &self,
-        info: Arc<AccessorInfo>,
-        path: &str,
-        operation: Operation,
-        error: Option<ErrorKind>,
-    ) -> Vec<KeyValue> {
+    fn create_attributes(&self, attrs: observe::MetricLabels) -> Vec<KeyValue> {
         let mut attributes = Vec::with_capacity(6);
 
         attributes.extend([
-            KeyValue::new(observe::LABEL_SCHEME, info.scheme().into_static()),
-            KeyValue::new(observe::LABEL_NAMESPACE, info.name().clone()),
-            KeyValue::new(observe::LABEL_ROOT, info.root().clone()),
-            KeyValue::new(observe::LABEL_OPERATION, operation.into_static()),
+            KeyValue::new(observe::LABEL_SCHEME, attrs.scheme.into_static()),
+            KeyValue::new(observe::LABEL_NAMESPACE, attrs.namespace),
+            KeyValue::new(observe::LABEL_ROOT, attrs.root),
+            KeyValue::new(observe::LABEL_OPERATION, attrs.operation),
         ]);
 
-        if let Some(path) = observe::path_label_value(path, self.path_label_level) {
-            attributes.push(KeyValue::new(observe::LABEL_PATH, path.to_owned()));
+        if let Some(error) = attrs.error {
+            attributes.push(KeyValue::new(observe::LABEL_ERROR, error.into_static()));
         }
 
-        if let Some(error) = error {
-            attributes.push(KeyValue::new(observe::LABEL_ERROR, error.into_static()));
+        if let Some(status_code) = attrs.status_code {
+            attributes.push(KeyValue::new(
+                observe::LABEL_STATUS_CODE,
+                status_code.as_u16() as i64,
+            ));
         }
 
         attributes
     }
 }
 
-fn exponential_boundary(start: f64, factor: f64, count: usize) -> Vec<f64> {
-    let mut boundaries = Vec::with_capacity(count);
-    let mut current = start;
-    for _ in 0..count {
-        boundaries.push(current);
-        current *= factor;
+/// Creates exponentially spaced histogram boundaries.
+///
+/// # Arguments
+///
+/// * `start` - The upper bound of the first boundary (must be positive)
+/// * `factor` - The factor to multiply each boundary by (must be greater than 1)
+/// * `count` - The number of boundaries to generate (must be at least 1)
+///
+/// # Panics
+///
+/// Panics if:
+/// * `start` is not positive
+/// * `factor` is not greater than 1
+/// * `count` is less than 1
+pub fn exponential_boundaries(start: f64, factor: f64, count: usize) -> Vec<f64> {
```

Comment:
Let's remove this API. This function is just an easy way to generate a bucket, and should not be a public API at the Layer level.